tests/callbacks_test.go

```diff
@@ -238,3 +241,122 @@ func (s *FunctionalSuite) TestWorkflowNexusCallbacks_CarriedOverContinueAsNew()
 		}
 	}
 }
+
+func (s *FunctionalSuite) TestWorkflowCallbacks_ForceActivityComplete() {
+	dc := s.testCluster.host.dcClient
+	dc.OverrideValue(s.T(), dynamicconfig.FrontendEnableCallbackAttachment, true)
+	defer dc.RemoveOverride(dynamicconfig.FrontendEnableCallbackAttachment)
+
+	sdkClient, err := client.Dial(client.Options{
+		HostPort:  s.testCluster.GetHost().FrontendGRPCAddress(),
+		Namespace: s.namespace,
+	})
+	s.NoError(err)
+
+	pp := temporalite.NewPortProvider()
+	callbackAddress := fmt.Sprintf("localhost:%d", pp.MustGetFreePort())
+	s.NoError(pp.Close())
+
+	ach := &serveHTTPHandler{
+		sdkClient:         sdkClient,
+		requestCompleteCh: make(chan error, 1),
+		activityErr:       make(chan error, 1),
+	}
+
+	mockErrorActivity := func(ctx context.Context) error {
+		ai := activity.GetInfo(ctx)
+		formData := url.Values{}
+		formData.Add("name_space", ai.WorkflowNamespace)
+		formData.Add("workflow_id", ai.WorkflowExecution.ID)
+		formData.Add("run_id", ai.WorkflowExecution.RunID)
+		formData.Add("activity_id", ai.ActivityID)
+
+		httpUrl := fmt.Sprintf("http://%s", callbackAddress)
+		_, err := http.PostForm(httpUrl, formData)
+		if err != nil {
+			panic(err)
+		}
+		err = errors.New("mock error")
+		ach.activityErr <- err
+		return err
+	}
+
+	wf := func(ctx workflow.Context) error {
+		ao := workflow.ActivityOptions{
+			StartToCloseTimeout: 3 * time.Minute,
+			RetryPolicy: &temporal.RetryPolicy{
+				InitialInterval: 2 * time.Minute,
+			},
+		}
+		ctx2 := workflow.WithActivityOptions(ctx, ao)
+		var err error
+		workflow.ExecuteActivity(ctx2, mockErrorActivity).Get(ctx2, &err)
+		return err
+	}
+
+	taskQueue := s.randomizeStr(s.T().Name())
+	workflowType := "test"
+	s.NoError(err)
+
+	shutdownServer := s.runCompletionHTTPServer(ach, callbackAddress)
+	defer func() {
+		err := shutdownServer()
+		if err != nil {
+			panic(err)
+		}
+	}()
```

Comment:
There's a better way to test this without an external HTTP server.

Reply:
May I get some reference for that? Thank you!

Reply:
Use the client directly from this test as I mention below after waiting on the describe response.